### PR TITLE
MySQL: Added SET TRANSACTION parsing

### PR DESCRIPTION
**Pull request by Xemptuous**
### Brief summary of the change made
Added various ways of parsing SET TRANSACTION statements in MySQL according to [the docs](https://dev.mysql.com/doc/refman/8.0/en/set-transaction.html)

### Are there any other side effects of this change that we should be aware of?
None that I am aware of.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 1]
```diff
@@ -2721,6 +2722,37 @@ class ReturnStatementSegment(BaseSegment):
...
+    As specified in https://dev.mysql.com/doc/refman/8.0/en/set-transaction.html
+    """
+
+    type = "set_transaction_statement"
+    match_grammar: Matchable = Sequence(
+        "SET",
+        OneOf("GLOBAL", "SESSION", optional=True),
+        "TRANSACTION",
+        Delimited(
+            AnySetOf(
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_mysql.py:2737]: Chaining Delimited and AnySetOf doesn't really work, as Delimited inherits from OneOf - can you remove the AnySetOf here?

Comment (Xemptuous) [reply src/sqlfluff/dialects/dialect_mysql.py:2737]: My intention was to only allow a max of 2 of these options, as you can't have more than 2 of these options at once. e.g., this won't work:
```sql
SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, READ WRITE;
```
I didn't see any args in `Delimiter()` to allow a max # of items.
Is there a better way to do this?

Comment (WittierDinosaur) [reply src/sqlfluff/dialects/dialect_mysql.py:2737]: Nah, there's no way to get what you're going for here cleanly. We'd need some kind of DelimitedSet operator, which we don't have at the mo

**Review by alanmcruickshank — APPROVED** [review]
Resolved PR feedback

**Pull request merged**